Clean-up in `computePositionedLogicalWidthUsing` to use shrink-to-fit in one step

Ahmad-S792 opened the pull request:
#### 6b0b9db9aa935654984d4e7c5b1e9128a8b90d0c
<pre>
Clean-up in `computePositionedLogicalWidthUsing` to use shrink-to-fit in one step

<a href="https://bugs.webkit.org/show_bug.cgi?id=279033">https://bugs.webkit.org/show_bug.cgi?id=279033</a>
<a href="https://rdar.apple.com/135156089">rdar://135156089</a>

Reviewed by Alan Baradlay.

This patch introduces `lambda` named `shrinkToFitLogicalWidth` to remove
duplicate logic to do `shrink-to-fit` in one go as clean-up.

* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::computePositionedLogicalWidthUsing const):

Canonical link: <a href="https://commits.webkit.org/283073@main">https://commits.webkit.org/283073@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e305b2600d3b504c55223c37217ca2b66d6317af

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/65132 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/44499 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/17746 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/69156 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/15738 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/67250 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/52282 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/16020 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/52324 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/10883 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/68198 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/41141 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/56378 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/32948 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/37812 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/13749 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/14614 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/59713 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/14088 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/70861 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/9084 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/13572 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/59654 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/9116 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/56438 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/59921 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14364 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/7505 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/1180 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/40311 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/41388 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/42569 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/41132 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/rendering/RenderBox.cpp:
```diff
@@ -4275,27 +4275,23 @@ void RenderBox::computePositionedLogicalWidthUsing(SizeType widthType, Length lo
 
         const LayoutUnit availableSpace = containerLogicalWidth - (marginLogicalLeftValue + marginLogicalRightValue + bordersPlusPadding);
 
+        auto shrinkToFitLogicalWidth = [this](LayoutUnit availableLogicalWidth, LayoutUnit bordersPlusPadding) -> LayoutUnit {
```
this probably should read
`auto shrinkToFitLogicalWidth = [&](auto availableLogicalWidth, auto bordersPlusPadding) {`

Ahmad-S792 replied:
Sure - can do it.

Ahmad-S792 replied:
Done in https://github.com/WebKit/WebKit/commit/e305b2600d3b504c55223c37217ca2b66d6317af.

alanbaradlay commented on Source/WebCore/rendering/RenderBox.cpp:
```diff
@@ -4275,27 +4275,23 @@ void RenderBox::computePositionedLogicalWidthUsing(SizeType widthType, Length lo
+        auto shrinkToFitLogicalWidth = [this](LayoutUnit availableLogicalWidth, LayoutUnit bordersPlusPadding) -> LayoutUnit {
+            LayoutUnit preferredLogicalWidth = maxPreferredLogicalWidth() - bordersPlusPadding;
+            LayoutUnit preferredMinLogicalWidth = minPreferredLogicalWidth() - bordersPlusPadding;
+            return std::clamp(availableLogicalWidth, preferredMinLogicalWidth, preferredLogicalWidth);
```
why not
`return std::clamp(availableLogicalWidth, preferredMinLogicalWidth, preferred`Max`LogicalWidth);`

Ahmad-S792 replied:
So - I rename `preferredLogicalWidth` to `preferredMaxLogicalWidth` and then update `return` as well. Can do it as well. :-)

Ahmad-S792 replied:
Done in https://github.com/WebKit/WebKit/commit/e305b2600d3b504c55223c37217ca2b66d6317af.

Ahmad-S792 committed:
Clean-up in `computePositionedLogicalWidthUsing` to use shrink-to-fit in one step

https://bugs.webkit.org/show_bug.cgi?id=279033
rdar://135156089

Reviewed by Alan Baradlay.

This patch introduces `lambda` named `shrinkToFitLogicalWidth` to remove
duplicate logic to do `shrink-to-fit` in one go as clean-up.

* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::computePositionedLogicalWidthUsing const):

Canonical link: https://commits.webkit.org/283073@main
